modules/ui/src/main/scala/lucuma/ui/forms/EnumSelect.scala

```diff
@@ -22,22 +23,25 @@ final case class EnumSelect[A](
   value:          Option[A],
   placeholder:    String = "",
   disabled:       Boolean = false,
-  onChange:       A => Callback = (_: A) => Callback.empty,
+  onChange:       A ==> Callback = ((_: A) => Callback.empty).reuseAlways,
   disabledItems:  Set[A] = Set.empty[A]
 )(implicit
   val enumerated: Enumerated[A],
-  val display:    Display[A]
-) extends ReactProps[EnumSelect[Any]](EnumSelect.component)
+  val display:    Display[A],
+  val reuse:      Reusability[A]
+) extends ReactProps[EnumSelect[Any]](EnumSelect.component) {
+  implicit protected val propsReuse: Reusability[EnumSelect[A]] = Reusability.derive
```

Comment:
Does this need to go here for the `derive` to work?

Reply:
We want the component to be type-parametrized on `A`, which must have a `Reusability` instance, but we want the component definition to be unique in a `val` so as not redefine it on every render.

`A` (and its `Reusability` instance) are not known until invocation, so the component's definition `shouldComponentUpdate` cannot rely on a general instance. I thought that placing it here and have the component's general reusability delegate to it could do the trick.